wallet-extension/README.md

```diff
@@ -0,0 +1,112 @@
+# Arweave Web Wallet Extensions
+
+This file demonstrates how an Arweave Web Wallet Extension should structure it's API so that all dApps, as well as `arweave-js` can use it properly. An implementation of this can be found at [th8ta/ArConnect](https://github.com/th8ta/ArConnect).
+
+## Logic of the wallet
+
+Arweave web wallet extensions are not just simple middlewares to sign transactions. They allow the user to interact with dApps in a more seamless way, similar to how Web 2 logins work.
+
+A wallet extension should provide specific permissions for the API calls that the user has to approve. To do this, the extension should show a popup window on *connect*, to ask the user for these permissions. Ideally the wallet should also provide some sort of limitation of how much the specific dApp can spend from the user's wallet, because the popup won't show on each transaction signing.
+
+Preferably the wallet should notify the user in an understandable way every time a transaction signing occurs.
+
+## The global API object
+
+All web wallet extensions should provide (inject) a field inside the browsers `window` object called `arweaveWallet`. This can be accessed via `window.arweaveWallet`. An example `d.ts` file of this object can be found in [injected-api.d.ts](injected-api.d.ts).
```

Comment:
```suggestion
All web wallet extensions should provide (inject) a field inside the browser's `window` object called `arweaveWallet`. This can be accessed via `window.arweaveWallet`. An example `d.ts` file of this object can be found in [injected-api.d.ts](injected-api.d.ts).
```